src/store/keycodes.js

```diff
@@ -13,7 +13,17 @@ import steno from './modules/keycodes/steno';
 const keycodePickerTabLayout = {
   ANSI_ISO: [...ansi, ...iso_jis],
   ISO_ANSI: [...iso_jis, ...ansi],
-  special: [...quantum, ...settings, ...media]
+  special: [...quantum, ...settings, ...media],
+  extra: Object.values(keymapExtras)
+    .map(({ keycodeLUT, prefix }) =>
+      Object.entries(keycodeLUT).map(([code, { name, title }]) => ({
+        code: title?.split(' ')[0], // split removes ' (dead)'
```

Comment:
Assuming that all `title`s follow the pattern `/\S+( \([^)]*\))?/` (i.e. a space-less token followed optionally by a parenthesized expression) is not a fair assumption to make.

Take for example the space cadet keycodes:
```
keymap_extras/keymap_french.js
173:  KC_COLN: { name: 'M', title: 'S(FR_M) (capital M)' },
174:  KC_PIPE: { name: 'µ', title: 'FR_MICR (capital µ)' },
176:  SC_LSPO: { name: 'LS / 9', title: 'Left Shift when held, 9 when tapped' },
177:  SC_RSPC: { name: 'RS / 0', title: 'Right Shift when held, 0 when tapped' },
178:  SC_LCPO: { name: 'LC / 9', title: 'Left Control when held, 9 when tapped' },
179:  SC_RCPC: { name: 'RC / 0', title: 'Right Control when held, 0 when tapped' },
180:  SC_LAPO: { name: 'LA / 9', title: 'Left Alt when held, 9 when tapped' },
181:  SC_RAPC: { name: 'RA / 0', title: 'Right Alt when held, 0 when tapped' },
183:  QK_GESC: {
184:    name: '²\nEsc',
185:    title: 'Esc normally, but ² when Shift or GUI is active'
186:  }
```

The `[extra.code, extra.name]` ends up looking like this:

```
Array [ "S(FR_SUP2)", "²" ]
Array [ "S(FR_M)", "M" ]
Array [ "FR_MICR", "µ" ]
Array [ "Left", "LS / 9" ]
Array [ "Right", "RS / 0" ]
Array [ "Left", "LC / 9" ]
Array [ "Right", "RC / 0" ]
Array [ "Left", "LA / 9" ]
Array [ "Right", "RA / 0" ]
Array [ "Esc", "²\nEsc" ]
```

You could argue that space cadet keycodes are out of scope so it does not matter what happens to them but the comment `removes ' (dead)'` can be misleading because `(dead)` is not the only parenthesized expression that can be found — (soft hyphen), (capital Å), (dead European symbol key), (한 Han ↔ 영 Yeong) and more...

If the goal is to build a LUT that can translate a default US keycode alias to a localized alias, scraping the data from the `title` field is not the best way to go. If need be, the `keymap_extras/keymap_###.js` can be easily regenerated with an extra field that can be dedicated to containing only the alias, without any extra information (like those seen in parentheses)

Reply:
Thanks for the clarification! As stated, I do not know this codebase, just thought it would be appreciated to create a MR with something that does solve a problem for me and is a registered issue.

Totally agree that we should have the correct data generated if that is an option, and that this is more correct:

```suggestion
        code: title?.split(' ')[0], // best-effort parsing of what may be a keycode
```

Reply:
Another thing to look out for are collisions in aliases. The most egregious case concerns `keymap_serbian` and `keymap_serbian_latin`. Both use the `RS_` prefix, however the former uses Cyrillic legends while the latter uses Latin legends. To illustrate, `RS_I` is displayed "И" if we assume "keymap_serbian" but the same keycode alias shows as "I" under "keymap_serbian_latin". It seems like the `keymap_serbian` data comes after the `keymap_serbian_latin` data in `keycodeLUT` so if a keymap.json file contains a `RS_###` keycode, it will always show the Cyrillic version.

The same problem also rears its head with the many Dvorak variations. `DV_RPRN`. In `keymap_dvorak` and `keymap_spanish_dvorak`, the expected legend is just ")" but in `keymap_dvorak_fr`, it should be "9 )" (i.e. pressing the key without modifiers produces ) but combining it with shift gives 9).

At the moment, I can't think of an elegant solution to this problem. For now, *if* the host_language field is present in the keymap.json, we can perhaps look to make sure that the data for that host_language is at the end of the `keycodeLUT` so that it takes priority over other possible matches.

Reply:
The collisions are intentional, only one header can be included at a time and thus only one set of "alternative" keycode names other than `KC_`. There's not much point in enabling multiple as they all send the same values over the wire anyway. These headers are solely for making your keymap match your OS keyboard language; the use case of having multiple layers for various languages doesn't make sense since this is not synced to/from the OS.

Basically, if there are (basic set) keycodes in the keymap that aren't `KC_` or can be found in the corresponding `host_language` header, this should be an error.